manticore/ethereum.py

```diff
@@ -1078,6 +1123,38 @@ def global_findings(self):
     def workspace(self):
         return self._executor._workspace._store.uri
 
+    def _concretize_offsets_and_sizes(self, state, signature, data):
```

Comment:
i'm having a hard time understanding this function. is it necessary so we can do `state.solve_one` on the arguments returned from ABI.parse() later on?
is there relevant ABI documentation on encoding of dynamic arguments we can link to here?

Reply:
Maybe this helps. Normally the ABI encoding looks like this

FUNCID, ARG0, ARG1, ARG2
sizeof(FUNCID) := 4
sizeof(ARG?) := 32

But if one of the arguments, let say ARG1, is a dynamic argument it uses more than 32 bytes. Then it'll be something like this 
```
FUNCID, ARG0, ARG1p, ARG2, ARG0N, ARG0.0,ARG00.1,,, ARG0.N
            \______________/\
sizeof(FUNCID) := 4
sizeof(ARG?) := 32
ARG1p is the offset where the actual ARG1 is in the input. Measured in bytes from the start of the arguments block. 
ARG0N is the actual size of ARG1 meassured in number of elements

ref. http://solidity.readthedocs.io/en/develop/abi-spec.html CTRL+F "Given the contract"
```